step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/execution/IsolatedAutomationPackageRepository.java

```
@@ -0,0 +1,116 @@
+
+    protected IsolatedAutomationPackageRepository() {
+        super(Set.of(REPOSITORY_PARAM_CONTEXTID));
+    }
+
+    @Override
+    public ArtefactInfo getArtefactInfo(Map<String, String> repositoryParameters) throws Exception {
+        AutomationPackage automationPackage = getAutomationPackageForContext(repositoryParameters);
+        if (automationPackage == null) {
+            return null;
+        }
+        ArtefactInfo info = new ArtefactInfo();
+        // TODO: type?
+        info.setType("automationPackage");
+        info.setName(automationPackage.getAttribute(AbstractOrganizableObject.NAME));
+        return info;
+    }
+
+    private AutomationPackage getAutomationPackageForContext(Map<String, String> repositoryParameters) {
+        // we expect, that there is only one automation package stored per context
+        AutomationPackageManager automationPackageManager = getContext(repositoryParameters);
+        return automationPackageManager.getAllAutomationPackages().findFirst().orElse(null);
+    }
+
+    private AutomationPackageManager getContext(Map<String, String> repositoryParameters) {
+        return inMemoryPackageManagers.get(repositoryParameters.get(REPOSITORY_PARAM_CONTEXTID));
+    }
+
+    @Override
+    public ImportResult importArtefact(ExecutionContext context, Map<String, String> repositoryParameters) {
+        AutomationPackageManager automationPackageManager = getContext(repositoryParameters);
+        AutomationPackage automationPackage = getAutomationPackageForContext(repositoryParameters);
+
+        String planId = repositoryParameters.get(RepositoryObjectReference.PLAN_ID);
+
+        Plan plan = automationPackageManager.getPackagePlans(automationPackage.getId().toString())
+                .stream()
+                .filter(p -> p.getId().toString().equals(planId)).findFirst().orElse(null);
+        if (plan == null) {
+            return null;
```

Should return an error in ImportResult